### PR TITLE
Error Prevention: Add index priority action

### PR DESCRIPTION
**Pull request by Angie-Zhang**
*Issue #, if available:*

https://github.com/opensearch-project/index-management/issues/646

*Description of changes:*

Add index priority action validation in Error Prevention framework

*CheckList:*
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/validation/ValidateIndexPriority.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    override fun execute(indexName: String): Validate {
+        // if these conditions are false, fail validation and do not execute index_priority action
+        if (hasReadOnlyAllowDeleteBlock(indexName)) {
+            return this
+        }
+        validationMessage = getValidationPassedMessage(indexName)
+        return this
+    }
+
+    fun hasReadOnlyAllowDeleteBlock(indexName: String): Boolean {
```
Comment (bowenlan-amzn) [inline src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/validation/ValidateIndexPriority.kt:40]: If this is re-used, then let's move to Validate base class

Comment (Angie-Zhang) [reply src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/validation/ValidateIndexPriority.kt:40]: To make review and testing easier, there will be a individual PR for all refactoring changes, including moving reusable methods & logs to abstract level.

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/validation/ValidateIndexPriority.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            return true
+        }
+        return false
+    }
+
+    @Suppress("TooManyFunctions")
+    companion object {
+        const val name = "validate_index_priority"
+        fun getReadOnlyAllowDeleteBlockMessage(index: String) = "read_only_allow_delete block is not null for index [index=$index]"
+        fun getValidationPassedMessage(index: String) = "Index Priority action validation passed for [index=$index]"
```
[the remark this answers is not in the document]
Comment (Angie-Zhang) [reply src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/validation/ValidateIndexPriority.kt:55]: To make review and testing easier, there will be a individual PR for all refactoring changes, including moving reusable methods & logs to abstract level.

**Review by bowenlan-amzn — APPROVED** [review]
For the to-be-refactor places, add a TODO refactor label in your next PR

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**